mmv1/products/alloydb/Cluster.yaml

```diff
@@ -193,6 +211,40 @@ properties:
           The initial password for the user.
         required: true
         sensitive: true
+  - !ruby/object:Api::Type::String
+    name: 'restoreSourceBackup'
+    ignore_read: true
+    immutable: true
+    conflicts:
+      - restore_source_cluster
+      - restore_point_in_time
+    description: |
+      The name of the backup that this cluster should be restored from. Conflicts with 'restore_source_cluster' and 'restore_point_in_time', both can't be set together.
+  - !ruby/object:Api::Type::String
+    name: 'restoreSourceCluster'
+    ignore_read: true
+    immutable: true
+    required_with:
+      - restore_point_in_time
+    conflicts:
+      - restore_source_backup
+    description: |
+      The name of the source cluster that this cluster should be restored from. The source cluster must have continuous
+      backup enabled for restore to succeed. Requires `restore_point_in_time` to also be set.
+
+      Conflicts with 'restore_source_backup', both can't be set together.
+  - !ruby/object:Api::Type::String
+    name: 'restorePointInTime'
+    ignore_read: true
+    immutable: true
+    required_with:
+      - restore_source_cluster
+    conflicts:
+      - restore_source_backup
+    description: |
+      The point in time that this cluster should be restored to, in RFC 3339 format. Requires `restore_source_cluster` to also be set.
+
+      Conflicts with 'restore_source_backup', both can't be set together.
```

Comment:
Hi Shuya, you are correct. This is the request object, and the backupSource and continuousBackupSource are what I am trying to send in the restore API. https://source.corp.google.com/depot/google3/google/cloud/alloydb/v1main/service.proto;l=1523-1532

I flattened it because I thought it looked better and was more intuitive than having two different NestedObjects. Especially with BackupSource, which only has one field. If that goes against your design principles, I'm happy to change this and add the nested objects.

Reply:
Thanks for the explanation. I'd suggest that we keep the same structure as the API, as I feel that it may be slightly easier to read. Also it may be easier to keep in sync if more fields are added in the future.